fineract-core/src/main/java/org/apache/fineract/infrastructure/core/service/database/JavaType.java

```diff
@@ -0,0 +1,352 @@
+package org.apache.fineract.infrastructure.core.service.database;
+
+import jakarta.validation.constraints.NotNull;
+import java.io.InputStream;
+import java.io.Reader;
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.fineract.infrastructure.core.exception.PlatformServiceUnavailableException;
+import org.apache.fineract.organisation.monetary.domain.MoneyHelper;
+import org.springframework.data.domain.Persistable;
+
+public enum JavaType {
```

Comment:
Just like I said with the SqlOperator class, this is way way way too complex for a single change. Let's shoot for the minimum you need instead of trying to solve all the world problems in a single breath.

Reply:
Well, this class is very useful and I could not solve this issue so quickly without this. Maybe you could a bit value what I did. This enum is not complex, I think it is very clear and easy to understand.

Reply:
I'm getting more and more convinced that we should use JOOQ. With below code we are getting closer and closer to create some kind of ORM layer...